### PR TITLE
[UI Components & Story] Add clickable video effect items

### PR DESCRIPTION
**Pull request by JamesBurnside**
# What
Add clickable video items that will be used in the background picker

# Why
To be used in the background picker in the video effects feature

# How Tested

- Can tab navigate over them and keyboard select them
- Aria labels correctly applied and aria disabled property is correctly applied when disabled

checkout the storybook link to play around with them

Screenshot showing None as selected, Blurred as hovered over and AddImage as disabled:
image

### CODE REVIEW

File: packages/react-components/src/components/VideoEffects/PresetVideoEffectsItems.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import React from 'react';
+import { _VideoEffectsItem, _VideoEffectsItemProps } from './VideoEffectsItem';
+
+/**
+ * 'None' Video Effects Item.
+ *
+ * @internal
+ */
+export const _VideoEffectsItemNone = (props: _VideoEffectsItemProps): JSX.Element => {
```
[the remark this answers is not in the document]
Comment (JamesBurnside) [reply packages/react-components/src/components/VideoEffects/PresetVideoEffectsItems.tsx:12]: I like no background 👍

File: packages/react-components/src/components/VideoEffects/VideoEffectsItem.tsx [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+          disabled
+            ? undefined
+            : (e) => {
+                if (e.key === 'Enter' || e.key === ' ') {
+                  props.onSelect?.(props.key);
+                }
+              }
+        }
+        tabIndex={props.disabled ? -1 : 0}
+        aria-label={props.title}
```
[the remark this answers is not in the document]
Comment (JamesBurnside) [reply packages/react-components/src/components/VideoEffects/VideoEffectsItem.tsx:165]: Good note. I put it in to make sure there was a aria-label but yeah title doesn't make sense to use as a title doesn't need to be supplied. I'll update this to be a required explicit field. Also made me realize this should be a aria-role button

[the remark this answers is not in the document]
Comment (JamesBurnside) [reply packages/react-components/src/components/VideoEffects/VideoEffectsItem.tsx]: Updated (added aria-label prop and role="button")

File: packages/react-components/src/components/VideoEffects/PresetVideoEffectsItems.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (JamesBurnside) [reply packages/react-components/src/components/VideoEffects/PresetVideoEffectsItems.tsx]: Update to `_VideoEffectItemNoBackground `

File: packages/react-components/src/components/VideoEffects/PresetVideoEffectsItems.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+/**
+ * 'None' Video Effects Item.
+ *
+ * @internal
+ */
+export const _VideoEffectsItemNoBackground = (props: _VideoEffectsItemProps): JSX.Element => {
+  const derivedProps = deriveProps(props, {
+    iconName: 'VideoEffectsNone',
+    title: 'None',
+    tooltipContent: 'Remove Background',
```
Comment (prabhjot-msft) [inline packages/react-components/src/components/VideoEffects/PresetVideoEffectsItems.tsx:16]: What if we don't have any effect enabled, would the remove background be right tooltip in that case?

File: packages/react-components/src/components/VideoEffects/PresetVideoEffectsItems.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    ariaLabel: 'Upload Background Image'
+  });
+
+  return <_VideoEffectsItem {...derivedProps} />;
+};
+
+/** Applies fallbacks if props were not specified */
+const deriveProps = (
+  props: _VideoEffectsItemProps,
+  fallbacks: {
```
[the remark this answers is not in the document]
Comment (JamesBurnside) [reply packages/react-components/src/components/VideoEffects/PresetVideoEffectsItems.tsx:58]: This is just an internal helper function for the components in this file to reduce all the components having duplicate code

File: packages/react-components/src/components/VideoEffects/VideoEffectsItem.tsx [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+     * The URL of the background image.
+     */
+    url: string;
+
+    /**
+     * The position of the background image.
+     * @default 'center'
+     * @see https://developer.mozilla.org/en-US/docs/Web/CSS/background-position
+     */
+    position?: string;
```
[the remark this answers is not in the document]
Comment (JamesBurnside) [reply packages/react-components/src/components/VideoEffects/VideoEffectsItem.tsx:87]: This is just if a developer using the picker item wants to change some of the styling of the background of this item.
Note: this is just for the picker item itself and not related to the background that is applied to the Video Tile.

(These are internal components that are not exported publicly so happy to remove if they cause confusion - is purely more flexibility for us)

File: packages/react-components/src/components/VideoEffects/VideoEffectsItem.tsx [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+
+    /**
+     * The size of the background image.
+     * @default 'cover'
+     * @see https://developer.mozilla.org/en-US/docs/Web/CSS/background-size
+     */
+    size?: string;
```
[the remark this answers is not in the document]
Comment (JamesBurnside) [reply packages/react-components/src/components/VideoEffects/VideoEffectsItem.tsx:94]: It's a great callout that these are confusing terms. I'll remove them given we're not using them

File: packages/react-components/src/components/VideoEffects/PresetVideoEffectsItems.tsx (hunk 3 above)
Comment (JamesBurnside) [reply packages/react-components/src/components/VideoEffects/PresetVideoEffectsItems.tsx:16]: Let's ask AlexP to include tooltips in his designs

File: packages/react-components/src/components/VideoEffects/VideoEffectsItem.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (prabhjot-msft) [reply packages/react-components/src/components/VideoEffects/VideoEffectsItem.tsx:87]: I think we are fine as these are mostly internal components and we control the behaviour. My concern around this was that does this `backgroundProp` has to match the type of `backgroundImage` (set of custom backgrounds passed by contoso) in that case we will need to expose these props to contoso as well.

[the remark this answers is not in the document]
Comment (JamesBurnside) [reply packages/react-components/src/components/VideoEffects/VideoEffectsItem.tsx:87]: Ah, yeah won't (and shouldn't) match. This UI layer should be astracted away from how its is used by the composite. The Composite SidePane will be mapping the backgrounds it gets passed in to the background picker and then handling the onSelectionChange events of the background picker

File: packages/react-components/src/components/VideoEffects/VideoEffectsItem.tsx [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+  /**
+   * Aria label for the Video effects item.
+   */
+  ariaLabel?: string;
+
+  /**
+   * Background to display for the Video effects item.
+   * @default undefined (no background image)
+   */
+  backgroundProps?: {
```
[the remark this answers is not in the document]
Comment (JamesBurnside) [reply packages/react-components/src/components/VideoEffects/VideoEffectsItem.tsx:76]: This isn't needed as this UI component doesn't use a specific background displayName, it has tooltip properties that likely will be the composite's backgrounds' displayNames

File: packages/react-components/src/components/VideoEffects/VideoEffectsItem.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (JamesBurnside) [reply packages/react-components/src/components/VideoEffects/VideoEffectsItem.tsx]: (removed position)

File: packages/react-components/src/components/VideoEffects/VideoEffectsItem.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (JamesBurnside) [reply packages/react-components/src/components/VideoEffects/VideoEffectsItem.tsx]: (removed size)

**Review by prabhjot-msft — APPROVED** [review]

**Pull request merged**